libs/common-services/src/main/java/uk/gov/di/ipv/core/library/tracing/TracingHttpClient.java

```diff
@@ -48,21 +48,49 @@ private TracingHttpClient(HttpClient baseClient) {
     }
 
     public static HttpClient newHttpClient() {
-        return new TracingHttpClient(
-                JavaHttpClientTelemetry.builder(GlobalOpenTelemetry.get())
-                        .build()
-                        .newHttpClient(HttpClient.newHttpClient()));
+        return new TracingHttpClient(getInstrumetedHttpClient());
     }
 
-    // Synchronize to prevent race conditions when multiple threads attempt to recreate client
-    private synchronized void recreateBaseClientIfNecessary() {
-        if (creationTime.plus(MAX_CLIENT_AGE).isBefore(Instant.now())) {
-            LOGGER.info("Recreating base HTTP client due to age");
-            this.baseClient = HttpClient.newHttpClient();
-            this.creationTime = Instant.now(); // Reset creation time
+    @Override
+    public <T> HttpResponse<T> send(
+            HttpRequest request, HttpResponse.BodyHandler<T> responseBodyHandler)
+            throws IOException, InterruptedException {
+        // PYIC-7590: Recreate the HTTP client, to avoid HTTP/2 connection issues.
+        // Check for client is 1hour old, and recreate if older.
+        recreateBaseClientIfNecessary();
+        try {
+            return sendWithTracing(request, responseBodyHandler);
+        } catch (IOException e) {
+            // We see occasional HTTP/2 GOAWAY messages from AWS when connections last
+            // for a long time (>1 hour). Retry them once, to recreate the connection.
+            // In the build environment we see connection resets for idle connections in
+            // the pool. Retrying uses a different connection.
+            if (HTTP_ERROR_MESSAGES_TO_RETRY.contains(e.getMessage())) {
```

Comment:
While you're here... one thing I spotted was that your change to include Connection Reset changed this from a substring match to an exact match, which means the GOAWAYs never actually get retried.

I'm not sure retrying helps very much (and recreating the client seems to pretty much eliminate the issue), but ideally we should either make the logic work or remove it.

(Doesn't have to be part of this PR)